stock_release_channel_partner_delivery_window/models/stock_release_channel.py

```diff
@@ -16,27 +22,83 @@ class StockReleaseChannel(models.Model):
         ),
     )
 
+    @delivery_date_generator(step="customer")
+    def _next_delivery_date_partner_delivery_window(self, delivery_date, partner):
+        """Get the next valid delivery date respecting customer delivery window.
+
+        The delivery date must be when the customer is open.
+        """
+        self.ensure_one()
+        partner.ensure_one()
+        if not self.respect_partner_delivery_time_windows:
+            while True:
+                delivery_date = yield delivery_date
```

Comment:
I can't understand the point of this, could you please explain?
This looks very complicated.